### ISSUE TITLE
ceph: Mons are allowing insecure global_id reclaim

### ISSUE THREAD

**Issue by psavva** [rook/rook#7746]
Bug Report

**Current Behaviour:**
ceph status return "HEALTH_WARN" 
with reason: mons are allowing insecure global_id reclaim

**Expected behavior:**
ceph status return "HEALTH_OK" 

**How to reproduce it (minimal and precise):**
Upgraded Ceph from 1.4 -> 1.5 -> 1.6.1
All OK.

Upgrade ceph using: https://rook.io/docs/rook/v1.6/ceph-upgrade.html#ceph-version-upgrades
to version: **ceph/ceph:v15.2.11-20210419**
```
NEW_CEPH_IMAGE='ceph/ceph:v15.2.11-20210419'
CLUSTER_NAME="$ROOK_CLUSTER_NAMESPACE"  # change if your cluster name is not the Rook namespace
kubectl -n $ROOK_CLUSTER_NAMESPACE patch CephCluster $CLUSTER_NAME --type=merge -p "{\"spec\": {\"cephVersion\": {\"image\": \"$NEW_CEPH_IMAGE\"}}}"
```
Once all was complete, all nodes upgraded OK.
Health warning started appearing from this point onwards.

**Environment**:
[root@FLATRO ceph]# cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

* Kernel (e.g. `uname -a`):
[root@FLATRO ceph]# uname -a
Linux FLATRO 3.10.0-1160.6.1.el7.x86_64 #1 SMP Tue Nov 17 13:59:11 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

* Cloud provider or hardware configuration:
Bare Metal. 3 Raw deviced for OSD
* Rook version (use `rook version` inside of a Rook Pod):
```
[root@FLATRO ceph]# kubectl -n $ROOK_OPERATOR_NAMESPACE get pod -o jsonpath='{range .items[*]}{.metadata.name}{"\n\t"}{.status.phase}{"\t\t"}{.spec.containers[0].image}{"\t"}{.spec.initContainers[0]}{"\n"}{end}' && \
> kubectl -n $ROOK_CLUSTER_NAMESPACE get pod -o jsonpath='{range .items[*]}{.metadata.name}{"\n\t"}{.status.phase}{"\t\t"}{.spec.containers[0].image}{"\t"}{.spec.initContainers[0].image}{"\n"}{end}'
csi-cephfsplugin-cv7bl
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-cephfsplugin-ln9kb
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-cephfsplugin-provisioner-777cdccd98-4r5qc
        Running         k8s.gcr.io/sig-storage/csi-attacher:v3.0.2
csi-cephfsplugin-provisioner-777cdccd98-k6mms
        Running         k8s.gcr.io/sig-storage/csi-attacher:v3.0.2
csi-cephfsplugin-w4lm5
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-c7tbs
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-hk4rz
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-krg8t
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-provisioner-6f8ddf6c9-h6z9w
        Running         k8s.gcr.io/sig-storage/csi-provisioner:v2.0.4
csi-rbdplugin-provisioner-6f8ddf6c9-zgf66
        Running         k8s.gcr.io/sig-storage/csi-provisioner:v2.0.4
prometheus-rook-prometheus-0
        Running         quay.io/prometheus/prometheus:v2.19.0
rook-ceph-crashcollector-davila-7c8dcdf5fb-nrmn2
        Running         ceph/ceph:v15.2.11-20210419     {"args":["/var/lib/ceph/crash/posted"],"command":["mkdir","-p"],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"make-container-crash-dir","resources":{},"securityContext":{"privileged":false},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/var/log/ceph","name":"rook-ceph-log"},{"mountPath":"/var/lib/ceph/crash","name":"rook-ceph-crash"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"default-token-vt6ws","readOnly":true}]}
rook-ceph-crashcollector-flatro-cf5484c79-j55cf
        Running         ceph/ceph:v15.2.11-20210419     {"args":["/var/lib/ceph/crash/posted"],"command":["mkdir","-p"],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"make-container-crash-dir","resources":{},"securityContext":{"privileged":false},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/var/log/ceph","name":"rook-ceph-log"},{"mountPath":"/var/lib/ceph/crash","name":"rook-ceph-crash"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"default-token-vt6ws","readOnly":true}]}
rook-ceph-crashcollector-roccas-b7d966868-qgpd8
        Running         ceph/ceph:v15.2.11-20210419     {"args":["/var/lib/ceph/crash/posted"],"command":["mkdir","-p"],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"make-container-crash-dir","resources":{},"securityContext":{"privileged":false},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/var/log/ceph","name":"rook-ceph-log"},{"mountPath":"/var/lib/ceph/crash","name":"rook-ceph-crash"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"default-token-vt6ws","readOnly":true}]}
rook-ceph-mgr-a-7949c4f996-qzknz
        Running         ceph/ceph:v15.2.11-20210419     {"args":["--verbose","--recursive","ceph:ceph","/var/log/ceph","/var/lib/ceph/crash","/var/lib/ceph/mgr/ceph-a"],"command":["chown"],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"chown-container-data-dir","resources":{},"securityContext":{"privileged":false},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/etc/ceph/keyring-store/","name":"rook-ceph-mgr-a-keyring","readOnly":true},{"mountPath":"/var/log/ceph","name":"rook-ceph-log"},{"mountPath":"/var/lib/ceph/crash","name":"rook-ceph-crash"},{"mountPath":"/var/lib/ceph/mgr/ceph-a","name":"ceph-daemon-data"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"rook-ceph-mgr-token-5ftx5","readOnly":true}]}
rook-ceph-mon-e-7ccf467765-l4nr5
        Running         ceph/ceph:v15.2.11-20210419     {"args":["--verbose","--recursive","ceph:ceph","/var/log/ceph","/var/lib/ceph/crash","/var/lib/ceph/mon/ceph-e"],"command":["chown"],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"chown-container-data-dir","resources":{},"securityContext":{"privileged":false},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/etc/ceph/keyring-store/","name":"rook-ceph-mons-keyring","readOnly":true},{"mountPath":"/var/log/ceph","name":"rook-ceph-log"},{"mountPath":"/var/lib/ceph/crash","name":"rook-ceph-crash"},{"mountPath":"/var/lib/ceph/mon/ceph-e","name":"ceph-daemon-data"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"default-token-vt6ws","readOnly":true}]}
rook-ceph-mon-h-5d5b8fdc65-89dqt
        Running         ceph/ceph:v15.2.11-20210419     {"args":["--verbose","--recursive","ceph:ceph","/var/log/ceph","/var/lib/ceph/crash","/var/lib/ceph/mon/ceph-h"],"command":["chown"],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"chown-container-data-dir","resources":{},"securityContext":{"privileged":false},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/etc/ceph/keyring-store/","name":"rook-ceph-mons-keyring","readOnly":true},{"mountPath":"/var/log/ceph","name":"rook-ceph-log"},{"mountPath":"/var/lib/ceph/crash","name":"rook-ceph-crash"},{"mountPath":"/var/lib/ceph/mon/ceph-h","name":"ceph-daemon-data"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"default-token-vt6ws","readOnly":true}]}
rook-ceph-mon-j-77f8f6d54d-kfqdl
        Running         ceph/ceph:v15.2.11-20210419     {"args":["--verbose","--recursive","ceph:ceph","/var/log/ceph","/var/lib/ceph/crash","/var/lib/ceph/mon/ceph-j"],"command":["chown"],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"chown-container-data-dir","resources":{},"securityContext":{"privileged":false},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/etc/ceph/keyring-store/","name":"rook-ceph-mons-keyring","readOnly":true},{"mountPath":"/var/log/ceph","name":"rook-ceph-log"},{"mountPath":"/var/lib/ceph/crash","name":"rook-ceph-crash"},{"mountPath":"/var/lib/ceph/mon/ceph-j","name":"ceph-daemon-data"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"default-token-vt6ws","readOnly":true}]}
rook-ceph-operator-789bbc898-2bw9q
        Running         rook/ceph:v1.6.1
rook-ceph-osd-0-5675cf8597-kqzlx
        Running         ceph/ceph:v15.2.11-20210419     {"command":["/bin/bash","-c","\nset -ex\n\nOSD_ID=0\nOSD_UUID=7aa990e4-7155-4ffe-9692-fbad2eedbe01\nOSD_STORE_FLAG=\"--bluestore\"\nOSD_DATA_DIR=/var/lib/ceph/osd/ceph-\"$OSD_ID\"\nCV_MODE=lvm\nDEVICE=/dev/ceph-c345acf0-7187-4443-9ac6-0aacbd93c669/osd-data-ad00327f-1b71-4457-8f6b-7e81a92eab2b\nMETADATA_DEVICE=\"$ROOK_METADATA_DEVICE\"\nWAL_DEVICE=\"$ROOK_WAL_DEVICE\"\n\n# active the osd with ceph-volume\nif [[ \"$CV_MODE\" == \"lvm\" ]]; then\n\tTMP_DIR=$(mktemp -d)\n\n\t# activate osd\n\tceph-volume \"$CV_MODE\" activate --no-systemd \"$OSD_STORE_FLAG\" \"$OSD_ID\" \"$OSD_UUID\"\n\n\t# copy the tmpfs directory to a temporary directory\n\t# this is needed because when the init container exits, the tmpfs goes away and its content with it\n\t# this will result in the emptydir to be empty when accessed by the main osd container\n\tcp --verbose --no-dereference \"$OSD_DATA_DIR\"/* \"$TMP_DIR\"/\n\n\t# unmount the tmpfs since we don't need it anymore\n\tumount \"$OSD_DATA_DIR\"\n\n\t# copy back the content of the tmpfs into the original osd directory\n\tcp --verbose --no-dereference \"$TMP_DIR\"/* \"$OSD_DATA_DIR\"\n\n\t# retain ownership of files to the ceph user/group\n\tchown --verbose --recursive ceph:ceph \"$OSD_DATA_DIR\"\n\n\t# remove the temporary directory\n\trm --recursive --force \"$TMP_DIR\"\nelse\n\tARGS=(--device ${DEVICE} --no-systemd --no-tmpfs)\n\tif [ -n \"$METADATA_DEVICE\" ]; then\n\t\tARGS+=(--block.db ${METADATA_DEVICE})\n\tfi\n\tif [ -n \"$WAL_DEVICE\" ]; then\n\t\tARGS+=(--block.wal ${WAL_DEVICE})\n\tfi\n\t# ceph-volume raw mode only supports bluestore so we don't need to pass a store flag\n\tceph-volume \"$CV_MODE\" activate \"${ARGS[@]}\"\nfi\n\n"],"env":[{"name":"CEPH_VOLUME_DEBUG","value":"1"},{"name":"CEPH_VOLUME_SKIP_RESTORECON","value":"1"},{"name":"DM_DISABLE_UDEV","value":"1"},{"name":"ROOK_CEPH_MON_HOST","valueFrom":{"secretKeyRef":{"key":"mon_host","name":"rook-ceph-config"}}},{"name":"CEPH_ARGS","value":"-m $(ROOK_CEPH_MON_HOST)"}],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"activate","resources":{},"securityContext":{"privileged":true},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/var/lib/ceph/osd/ceph-0","name":"activate-osd"},{"mountPath":"/dev","name":"devices"},{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"rook-ceph-osd-token-sgjsv","readOnly":true}]}
rook-ceph-osd-1-7f6c85564f-7j54v
        Running         ceph/ceph:v15.2.11-20210419     {"command":["/bin/bash","-c","\nset -ex\n\nOSD_ID=1\nOSD_UUID=408cb269-99d0-433d-bd3a-1ea2e533527f\nOSD_STORE_FLAG=\"--bluestore\"\nOSD_DATA_DIR=/var/lib/ceph/osd/ceph-\"$OSD_ID\"\nCV_MODE=lvm\nDEVICE=/dev/ceph-7e7c9f99-f313-4da5-b36d-c5c26cbb50e5/osd-data-da2d61c8-6ef8-4acf-a655-a3ebd59b2e9b\nMETADATA_DEVICE=\"$ROOK_METADATA_DEVICE\"\nWAL_DEVICE=\"$ROOK_WAL_DEVICE\"\n\n# active the osd with ceph-volume\nif [[ \"$CV_MODE\" == \"lvm\" ]]; then\n\tTMP_DIR=$(mktemp -d)\n\n\t# activate osd\n\tceph-volume \"$CV_MODE\" activate --no-systemd \"$OSD_STORE_FLAG\" \"$OSD_ID\" \"$OSD_UUID\"\n\n\t# copy the tmpfs directory to a temporary directory\n\t# this is needed because when the init container exits, the tmpfs goes away and its content with it\n\t# this will result in the emptydir to be empty when accessed by the main osd container\n\tcp --verbose --no-dereference \"$OSD_DATA_DIR\"/* \"$TMP_DIR\"/\n\n\t# unmount the tmpfs since we don't need it anymore\n\tumount \"$OSD_DATA_DIR\"\n\n\t# copy back the content of the tmpfs into the original osd directory\n\tcp --verbose --no-dereference \"$TMP_DIR\"/* \"$OSD_DATA_DIR\"\n\n\t# retain ownership of files to the ceph user/group\n\tchown --verbose --recursive ceph:ceph \"$OSD_DATA_DIR\"\n\n\t# remove the temporary directory\n\trm --recursive --force \"$TMP_DIR\"\nelse\n\tARGS=(--device ${DEVICE} --no-systemd --no-tmpfs)\n\tif [ -n \"$METADATA_DEVICE\" ]; then\n\t\tARGS+=(--block.db ${METADATA_DEVICE})\n\tfi\n\tif [ -n \"$WAL_DEVICE\" ]; then\n\t\tARGS+=(--block.wal ${WAL_DEVICE})\n\tfi\n\t# ceph-volume raw mode only supports bluestore so we don't need to pass a store flag\n\tceph-volume \"$CV_MODE\" activate \"${ARGS[@]}\"\nfi\n\n"],"env":[{"name":"CEPH_VOLUME_DEBUG","value":"1"},{"name":"CEPH_VOLUME_SKIP_RESTORECON","value":"1"},{"name":"DM_DISABLE_UDEV","value":"1"},{"name":"ROOK_CEPH_MON_HOST","valueFrom":{"secretKeyRef":{"key":"mon_host","name":"rook-ceph-config"}}},{"name":"CEPH_ARGS","value":"-m $(ROOK_CEPH_MON_HOST)"}],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"activate","resources":{},"securityContext":{"privileged":true},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/var/lib/ceph/osd/ceph-1","name":"activate-osd"},{"mountPath":"/dev","name":"devices"},{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"rook-ceph-osd-token-sgjsv","readOnly":true}]}
rook-ceph-osd-2-59999c7cf9-55xqw
        Running         ceph/ceph:v15.2.11-20210419     {"command":["/bin/bash","-c","\nset -ex\n\nOSD_ID=2\nOSD_UUID=9e41a8f1-a989-44c8-9aaa-89c9f32d5f09\nOSD_STORE_FLAG=\"--bluestore\"\nOSD_DATA_DIR=/var/lib/ceph/osd/ceph-\"$OSD_ID\"\nCV_MODE=lvm\nDEVICE=/dev/ceph-bba046f6-836e-45ea-a669-d3d24c04724c/osd-data-fd9ed982-c2a3-47e8-9389-2dbb63294625\nMETADATA_DEVICE=\"$ROOK_METADATA_DEVICE\"\nWAL_DEVICE=\"$ROOK_WAL_DEVICE\"\n\n# active the osd with ceph-volume\nif [[ \"$CV_MODE\" == \"lvm\" ]]; then\n\tTMP_DIR=$(mktemp -d)\n\n\t# activate osd\n\tceph-volume \"$CV_MODE\" activate --no-systemd \"$OSD_STORE_FLAG\" \"$OSD_ID\" \"$OSD_UUID\"\n\n\t# copy the tmpfs directory to a temporary directory\n\t# this is needed because when the init container exits, the tmpfs goes away and its content with it\n\t# this will result in the emptydir to be empty when accessed by the main osd container\n\tcp --verbose --no-dereference \"$OSD_DATA_DIR\"/* \"$TMP_DIR\"/\n\n\t# unmount the tmpfs since we don't need it anymore\n\tumount \"$OSD_DATA_DIR\"\n\n\t# copy back the content of the tmpfs into the original osd directory\n\tcp --verbose --no-dereference \"$TMP_DIR\"/* \"$OSD_DATA_DIR\"\n\n\t# retain ownership of files to the ceph user/group\n\tchown --verbose --recursive ceph:ceph \"$OSD_DATA_DIR\"\n\n\t# remove the temporary directory\n\trm --recursive --force \"$TMP_DIR\"\nelse\n\tARGS=(--device ${DEVICE} --no-systemd --no-tmpfs)\n\tif [ -n \"$METADATA_DEVICE\" ]; then\n\t\tARGS+=(--block.db ${METADATA_DEVICE})\n\tfi\n\tif [ -n \"$WAL_DEVICE\" ]; then\n\t\tARGS+=(--block.wal ${WAL_DEVICE})\n\tfi\n\t# ceph-volume raw mode only supports bluestore so we don't need to pass a store flag\n\tceph-volume \"$CV_MODE\" activate \"${ARGS[@]}\"\nfi\n\n"],"env":[{"name":"CEPH_VOLUME_DEBUG","value":"1"},{"name":"CEPH_VOLUME_SKIP_RESTORECON","value":"1"},{"name":"DM_DISABLE_UDEV","value":"1"},{"name":"ROOK_CEPH_MON_HOST","valueFrom":{"secretKeyRef":{"key":"mon_host","name":"rook-ceph-config"}}},{"name":"CEPH_ARGS","value":"-m $(ROOK_CEPH_MON_HOST)"}],"image":"ceph/ceph:v15.2.11-20210419","imagePullPolicy":"IfNotPresent","name":"activate","resources":{},"securityContext":{"privileged":true},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/var/lib/ceph/osd/ceph-2","name":"activate-osd"},{"mountPath":"/dev","name":"devices"},{"mountPath":"/etc/ceph","name":"rook-config-override","readOnly":true},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"rook-ceph-osd-token-sgjsv","readOnly":true}]}
rook-ceph-osd-prepare-davila-cg99v
        Succeeded               ceph/ceph:v15.2.11-20210419     {"args":["copy-binaries","--copy-to-dir","/rook"],"image":"rook/ceph:v1.5.10","imagePullPolicy":"IfNotPresent","name":"copy-bins","resources":{},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/rook","name":"rook-binaries"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"rook-ceph-osd-token-sgjsv","readOnly":true}]}
rook-ceph-osd-prepare-flatro-xw8vn
        Succeeded               ceph/ceph:v15.2.11-20210419     {"args":["copy-binaries","--copy-to-dir","/rook"],"image":"rook/ceph:v1.5.10","imagePullPolicy":"IfNotPresent","name":"copy-bins","resources":{},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/rook","name":"rook-binaries"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"rook-ceph-osd-token-sgjsv","readOnly":true}]}
rook-ceph-osd-prepare-roccas-pfvkd
        Succeeded               ceph/ceph:v15.2.11-20210419     {"args":["copy-binaries","--copy-to-dir","/rook"],"image":"rook/ceph:v1.5.10","imagePullPolicy":"IfNotPresent","name":"copy-bins","resources":{},"terminationMessagePath":"/dev/termination-log","terminationMessagePolicy":"File","volumeMounts":[{"mountPath":"/rook","name":"rook-binaries"},{"mountPath":"/var/run/secrets/kubernetes.io/serviceaccount","name":"rook-ceph-osd-token-sgjsv","readOnly":true}]}
rook-ceph-tools-6f94f7d446-pgm8r
        Running         rook/ceph:v1.5.10
rook-discover-jmb9l
        Running         rook/ceph:v1.6.1
rook-discover-nbxr8
        Running         rook/ceph:v1.6.1
rook-discover-s88j9
        Running         rook/ceph:v1.6.1
csi-cephfsplugin-cv7bl
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-cephfsplugin-ln9kb
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-cephfsplugin-provisioner-777cdccd98-4r5qc
        Running         k8s.gcr.io/sig-storage/csi-attacher:v3.0.2
csi-cephfsplugin-provisioner-777cdccd98-k6mms
        Running         k8s.gcr.io/sig-storage/csi-attacher:v3.0.2
csi-cephfsplugin-w4lm5
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-c7tbs
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-hk4rz
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-krg8t
        Running         k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.0.1
csi-rbdplugin-provisioner-6f8ddf6c9-h6z9w
        Running         k8s.gcr.io/sig-storage/csi-provisioner:v2.0.4
csi-rbdplugin-provisioner-6f8ddf6c9-zgf66
        Running         k8s.gcr.io/sig-storage/csi-provisioner:v2.0.4
prometheus-rook-prometheus-0
        Running         quay.io/prometheus/prometheus:v2.19.0
rook-ceph-crashcollector-davila-7c8dcdf5fb-nrmn2
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-crashcollector-flatro-cf5484c79-j55cf
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-crashcollector-roccas-b7d966868-qgpd8
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-mgr-a-7949c4f996-qzknz
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-mon-e-7ccf467765-l4nr5
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-mon-h-5d5b8fdc65-89dqt
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-mon-j-77f8f6d54d-kfqdl
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-operator-789bbc898-2bw9q
        Running         rook/ceph:v1.6.1
rook-ceph-osd-0-5675cf8597-kqzlx
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-osd-1-7f6c85564f-7j54v
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-osd-2-59999c7cf9-55xqw
        Running         ceph/ceph:v15.2.11-20210419     ceph/ceph:v15.2.11-20210419
rook-ceph-osd-prepare-davila-cg99v
        Succeeded               ceph/ceph:v15.2.11-20210419     rook/ceph:v1.5.10
rook-ceph-osd-prepare-flatro-xw8vn
        Succeeded               ceph/ceph:v15.2.11-20210419     rook/ceph:v1.5.10
rook-ceph-osd-prepare-roccas-pfvkd
        Succeeded               ceph/ceph:v15.2.11-20210419     rook/ceph:v1.5.10
rook-ceph-tools-6f94f7d446-pgm8r
        Running         rook/ceph:v1.5.10
rook-discover-jmb9l
        Running         rook/ceph:v1.6.1
rook-discover-nbxr8
        Running         rook/ceph:v1.6.1
rook-discover-s88j9
        Running         rook/ceph:v1.6.1

```

* Storage backend version (e.g. for ceph do `ceph -v`): 
v15.2.11-20210419

* Kubernetes version (use `kubectl version`):
[root@FLATRO ceph]# kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:17:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:41:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
Bare Metal
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):
[root@FLATRO ceph]# kubectl -n $ROOK_CLUSTER_NAMESPACE exec -it $TOOLS_POD -- ceph status
```
  cluster:
    id:     3bd7a445-c9b4-4524-b298-b96d25e09251
    health: HEALTH_WARN
            mons are allowing insecure global_id reclaim

  services:
    mon: 3 daemons, quorum e,h,j (age 56m)
    mgr: a(active, since 55m)
    osd: 3 osds: 3 up (since 106m), 3 in (since 2w)

  data:
    pools:   2 pools, 33 pgs
    objects: 4.51k objects, 15 GiB
    usage:   47 GiB used, 853 GiB / 900 GiB avail
    pgs:     33 active+clean

  io:
    client:   69 KiB/s wr, 0 op/s rd, 5 op/s wr

```

**Comment by mosesdd** [comment]
this seems to be related to
https://docs.ceph.com/en/latest/security/CVE-2021-20288/

**Comment by psavva** [comment]
Correct, this must be handled in rook-ceph I guess

**Comment by travisn** [comment]
I can confirm this is happening in all clusters, whether a clean install or upgraded cluster, running at least versions: v14.2.20, v15.2.11 or v16.2.1. 

According to the [CVE](https://docs.ceph.com/en/latest/security/CVE-2021-20288/) also previously mentioned, there is a security issue where clients need to be upgraded to the releases mentioned. Once all the clients are updated (e.g. the rook daemons and csi driver), a new setting needs to be applied to the cluster that will disable allowing the insecure mode. 

If you see both these health warnings, then either one of the rook or csi daemons has not been upgraded yet, or some other client is detected on the older version:
```
    health: HEALTH_WARN
            client is using insecure global_id reclaim
            mon is allowing insecure global_id reclaim
```

If you only see this one warning, then the insecure mode should be disabled:
```
    health: HEALTH_WARN
            mon is allowing insecure global_id reclaim
```

To disable the insecure mode from the toolbox after all the clients are upgraded:
```
ceph config set mon mon_warn_on_insecure_global_id_reclaim_allowed false
```

Rook could set this flag automatically after the clients have all been updated.

**Comment by psavva** [comment]
Thank you for your analysis and confirmation.
I will proceed with the update on my cluster as suggested.

**Comment by cyclops8456** [comment]
@psavva @travisn Hit the same issue earlier on a test cluster, setting `ceph config set mon mon_warn_on_insecure_global_id_reclaim_allowed false` resulted in the toolbox losing connectivity to the cluster (and presumably the operator as well).  Not sure if they need the ceph client used in those images updating as well.

**Comment by cyclops8456** [comment]
Adding this setting to the `rook-config-override` configmap resulted in the cluster not coming up at all (stuck at configuring mons); configmap applied before applying the cluster yaml.

**Comment by cyclops8456** [comment]
@psavva @travisn Further to the comments from yesterday, flattened everything to confirm, and applying the configmap below looks to resolve the issue for 1.6.x but breaks 1.5.x.  If this configmap is applied before the cluster yaml is applied then the monitors get this setting; applying it after will require the pods to be deleted / restarted.

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: rook-config-override
  namespace: rook-ceph
data:
  config: |
    [mon]
    mon_warn_on_insecure_global_id_reclaim_allowed = false
```

**Comment by cyclops8456** [comment]
@psavva @travisn Further to the above, adding the configmap above to 1.5.x resolves the issue there as well; there must have been something I didn't completely remove during testing of the above changes.

**Comment by cyclops8456** [comment]
@psavva @travisn Looks like I was hasty in saying the above configmap resolves the issue for 1.5.x after all (previous comment removed); I should have left the setup for longer.  It appears the client needs to be updated in the 1.5.10 operator, as `ceph status` is reporting `clients are using insecure global_id reclaim`

**Comment by psavva** [comment]
I had updated to 1.5.10 and also set the flag as suggested.  HEALTH OK after that.

**Comment by cyclops8456** [comment]
@psavva I had that initially, which is why I thought it was working as expected, but as soon as I provisioned a PVC, it showed up with `clients are using insecure global_id reclaim`.  Presuming it was triggered by the CSI plugins, but haven't been able to check further.

**Comment by travisn** [comment]
@psavva @cyclops8456 On a test cluster where I had rook v1.6.1, ceph pacific v16.2.1, and the latest csi driver v3.3.1, disabling the insecure global_id resulted in the cluster working fine. I was able to create new PVs, run the toolbox commands, and the operator was functioning as expected.

Note that the toolbox is not automatically upgraded, so if the toolbox commands are failing after disabling the insecure global_id, you may have an older toolbox so you may need to update the toolbox.

On Rook v1.5.x, the CSI driver deployed by default is v3.2, which would not have the latest Ceph base image with the secure global_id fix. @Madhu-1 @humblec Is there a newer ceph-csi v3.2.x release that would pick this up so we can support the secure global_id on Rook v1.5.x?

**Comment by Madhu-1** [comment]
@travisn sorry I missed the ping. we don't have one yet. we can make a new cephcsi v3.2.x release with the updated base image.

**Comment by humblec** [comment]
> @psavva @cyclops8456 On a test cluster where I had rook v1.6.1, ceph pacific v16.2.1, and the latest csi driver v3.3.1, disabling the insecure global_id resulted in the cluster working fine. I was able to create new PVs, run the toolbox commands, and the operator was functioning as expected.
> 
> Note that the toolbox is not automatically upgraded, so if the toolbox commands are failing after disabling the insecure global_id, you may have an older toolbox so you may need to update the toolbox.
> 
> On Rook v1.5.x, the CSI driver deployed by default is v3.2, which would not have the latest Ceph base image with the secure global_id fix. @Madhu-1 @humblec Is there a newer ceph-csi v3.2.x release that would pick this up so we can support the secure global_id on Rook v1.5.x?

The CVE fix has only made into 3.3.1 release till now.  If there is a compatibility issue, we have to look into the possibility of a release in 3.2.x.  Have to check the ceph compatibility against 3.2 branch though.

**Comment by idryomov** [comment]
> To disable the insecure mode from the toolbox after all the clients are upgraded:
> 
> ```
> ceph config set mon mon_warn_on_insecure_global_id_reclaim_allowed false
> ```
> 
> Rook could set this flag automatically after the clients have all been updated.

That would disable the warning about insecure reclaim.  `mon_warn_*` settings should stay on.

To disallow insecure reclaim you want:

```
ceph config set mon auth_allow_insecure_global_id_reclaim false
```

but only after *everything* has been upgraded, including all toolbox pods etc.

**Comment by travisn** [comment]
@idryomov Right, thanks for the correction, I'll update the original comment so the wrong command isn't found

**Comment by cyclops8456** [comment]
> The CVE fix has only made into 3.3.1 release till now. If there is a compatibility issue, we have to look into the possibility of a release in 3.2.x. Have to check the ceph compatibility against 3.2 branch though.

This has been added to the 3.2.2 release, so it looks to be resolved for the 1.5 branch now, providing the operator and toolbox are updated to use the 1.5.11 image, and then the cluster CR is updated to use the 15.2.11 image.  @travisn This can probably be closed now?

> Note that the toolbox is not automatically upgraded, so if the toolbox commands are failing after disabling the insecure global_id, you may have an older toolbox so you may need to update the toolbox.

@travisn Yeah, I must have missed updating one of the image versions; PEBKAC on this one, while bouncing back and forth between 1.5 and 1.6 branches.

**Comment by cyclops8456** [comment]
Connected to the earlier comment on the configmap, this will now work (providing all pods are been upgraded to the latest version before it's applied; or starting from scratch from the latest images).

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: rook-config-override
  namespace: rook-ceph
data:
  config: |
    [mon]
    auth_allow_insecure_global_id_reclaim = false
```
It will require the mon pods to be restarted if it's applied when they're already running.

**Ensure everything has been updated to the latest version before applying the above**

**Comment by travisn** [comment]
Running this command from the toolbox will have the same effect as well, but may be simpler for the one-time action instead of maintaining the configmap:
```
ceph config set mon auth_allow_insecure_global_id_reclaim false
```

**Comment by cyclops8456** [comment]
Agreed.  Looking through the documentation, this does seem the way to go.  It would be nice to have this working out of the box without having to do that, but a one time set of this is easier to do than maintain the configmap; as you've said.

**Comment by uhthomas** [comment]
For anyone who decides to fix this using a ConfigMap, ensure your config is suffixed with a new line. `init-mon-fs` gets very unhappy otherwise and will fail with

```
global_init: error reading config file.
```

https://github.com/uhthomas/automata/commit/f0da04221bb3c73571bb2407db79ee5acb877040

**Comment by travisn** [comment]
Note that if a client such as the toolbox attempts to connect after the insecure global ID has been disabled, it will result in an error like this:
```
2021-06-09T04:07:27.780+0000 7f9892d9d700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2,1]
```

**Comment by leseb** [comment]
> Note that if a client such as the toolbox attempts to connect after the insecure global ID has been disabled, it will result in an error like this:
> 
> ```
> 2021-06-09T04:07:27.780+0000 7f9892d9d700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2,1]
> ```

Are you sure? Typically it means we are using an incorrect keyring but it could be another instance of it since this message is far from being clear...

**Comment by idryomov** [comment]
Yes, unfortunately this is the way it is. The monitor returns the same error code for most authentication failures and all the client can do is print the list of allowed protocols (`none`/`1` and `cephx`/`2`).  The real reason is buried in the monitor logs.

I think it could be improved to at least not give the impression that the protocol is somehow not supported.  When `none` is disabled, it will be `handle_auth_bad_method server allowed_methods [2] but i only support [2]` -- even more confusing...

**Comment by electrical** [comment]
@idryomov on advice of @travisn in https://rook-io.slack.com/archives/CK9CF5H2R/p1623252358306800 i'm pinging you here.
I'm trying an upgrade from 15.2.11 to 15.2.13 but running into the issue that i get the error:
```monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2,1]```

Currently the operator pod isn't progressing with anything.

```
2021-06-09 17:54:05.345751 E \| op-mon: attempting to continue after failing to start mon "e". failed to update mon deployment rook-ceph-mon-e: failed to check if deployment "rook-ceph-mon-e" can be updated. failed to check if we can stop the deployment rook-ceph-mon-e: failed to get ceph daemons versions: failed to run 'ceph versions'. 2021-06-09T17:54:05.337+0000 7f2baa59c700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2,1]
--
Wed, Jun 9 2021 6:54:05 pm | 2021-06-09T17:54:05.337+0000 7f2baad9d700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2,1]
Wed, Jun 9 2021 6:54:05 pm | 2021-06-09T17:54:05.337+0000 7f2ba9d9b700 -1 monclient(hunting): handle_auth_bad_method server allowed_methods [2] but i only support [2,1]
Wed, Jun 9 2021 6:54:05 pm | [errno 13] RADOS permission denied (error connecting to the cluster)
```

**Comment by electrical** [comment]
After some digging with @travisn we found it's not related to the insecure global_id part.
Somehow the local cluster admin keyring is corrupt while the external cluster can still connect to it fine. the Ceph dashboard is working as well.

**Comment by travisn** [comment]
With #8089 planned for v1.6.5, the operator will disable the insecure global ID connections automatically. If there are any insecure clients that weren't online at the time, they not be able to connect to the cluster when they come back online. 

If you need to re-enable the insecure connections temporarily until all clients are updated, see the [CVE](https://docs.ceph.com/en/latest/security/CVE-2021-20288/) for instructions on the following:
1. Enable the insecure connections. 
   - For example: `ceph config set mon auth_expose_insecure_global_id_reclaim false`
   - Note that the toolbox must be updated to a recent release or else toolbox commands will also fail.
1. Disable the health warning that indicates insecure connections are allowed. The operator will only disable the insecure connections if it sees this warning when no insecure clients are connected. 
   - For example: `ceph health mute AUTH_INSECURE_GLOBAL_ID_RECLAIM_ALLOWED 1w`

**Comment by idryomov** [comment]
This won't work.  Setting `auth_expose_insecure_global_id_reclaim` to false would only *appear* to enable the insecure connections.  The old client would be allowed in initially, but when it attempts to reclaim its global ID after a monitor restart or some kind of network disruption, it would be denied and eventually hang in a very weird state.

All `auth_expose_insecure_global_id_reclaim` does is it runs the clients through the reclaim procedure on the initial connect, to make the old clients fail fast and safe.  The actual enforcement is controlled by `auth_allow_insecure_global_id_reclaim`.

`auth_expose_insecure_global_id_reclaim` defaults to true and should virtually never be changed.  The linked CVE page mentions it in `Fix details`, not `Recommendations`.  In an attempt to explain the technical side of the fix, we seem to have emphasised it too much.  In retrospect, `auth_expose_insecure_global_id_reclaim` should probably have been a hidden undocumented option.

**Comment by travisn** [comment]
@idryomov  For step 1 I intended to document that `auth_allow_insecure_global_id_reclaim` would be set back to the default value of true. That should working, right?

**Comment by idryomov** [comment]
Yes, as long as `configureHealthSettings()` doesn't set it back to false the next time it runs...

**Comment by travisn** [comment]
With #8089 included in v1.6.5, the operator will disable the insecure global ID connections automatically. If there are any insecure clients that weren't online at the time, they not be able to connect to the cluster when they come back online. 

If you need to re-enable the insecure connections temporarily until all clients are updated, see the [CVE](https://docs.ceph.com/en/latest/security/CVE-2021-20288/) for instructions on the following:
1. Disable the health warning that indicates insecure connections are allowed. The operator will only disable the insecure connections if it sees this warning when no insecure clients are connected. 
   - In the toolbox: `ceph health mute AUTH_INSECURE_GLOBAL_ID_RECLAIM_ALLOWED 1w`
   - The health warning can also be disabled permanently (see the [CVE](https://docs.ceph.com/en/latest/security/CVE-2021-20288/)), though it is not recommended. 
   - If the health warning suppression expires, the operator will again disable the insecure connections if it does not detect any active.
1. Enable the insecure connections:
   - In the toolbox: `ceph config set mon auth_allow_insecure_global_id_reclaim true`
   - Note that the toolbox must be updated to a recent release or else toolbox commands will also fail.

**Comment by RyuunoAelia** [comment]
I had this exact issue when updating my ceph cluster from 1.6.0 to 1.6.5 using rook-ceph 1.6.7. From what I have seen in the logs it tried to first disable the global_id and then restarted the OSD.0 which failed to restart. Enabling back the feature without silencing the warning seems to have fixed the issue.

**Comment by thobib** [comment]
Thanks for Information

**Comment by thenonsequitur** [comment]
In case it helps anyone else in the future, I just ran into this warning after a rook upgrade, and for me all I had to do was restart all the csi plugin pods and the warning went away.

**Comment by zhangdev200** [comment]
I runned this command"config set mon auth_allow_insecure_global_id_reclaim false". But it shows 
"Invalid command: unused arguments: [u'false']
config set <key> <value> :  Set a configuration option at runtime (not persistent)
Error EINVAL: invalid command"

**Comment by zhangdev200** [comment]
> Running this command from the toolbox will have the same effect as well, but may be simpler for the one-time action instead of maintaining the configmap:
> 
> ```
> ceph config set mon auth_allow_insecure_global_id_reclaim false
> ```

I runned this command"config set mon auth_allow_insecure_global_id_reclaim false". But it shows
"Invalid command: unused arguments: [u'false']
config set : Set a configuration option at runtime (not persistent)
Error EINVAL: invalid command"

**Comment by travisn** [comment]
@zhangdev200 What version of Rook and version of Ceph do you have? For older versions of ceph, the syntax may be slightly different.

**Comment by agaoglu** [comment]
While upgrading from v14 to v16 directly, this sometimes breaks the cluster. IIUC from the code, rook checks for two mon health-checks and if there's no client check while there's an allowed one, it assumes all clients are up-to-date and goes ahead with disabling the setting. But - sometimes - the allowed check fires right after mon-a is upgraded, without the clients check. Rook does its thing and in a few seconds, entire cluster is down because no osds have been upgraded yet. And of-course rook cannot continue upgrading osds. At this point it sometimes works to re-enable globalid reclaim manually, but sometimes it becomes a race with the operator, disabling it again, all with a down cluster.

I'm not sure if there's an ordering for these checks to appear on ceph-side but I think it is at-least theoretically possible since one is just a config check, but the other one requires some sort of communication with a client. The activity on the cluster might also have an effect on this possibility. I'm trying to upgrade test clusters and they sometimes have very little or no traffic.

**Comment by idryomov** [comment]
> I'm not sure if there's an ordering for these checks to appear on ceph-side but I think it is at-least theoretically possible since one is just a config check, but the other one requires some sort of communication with a client.

No communication is required per se, it's just that clients can be connected to different monitors.  An upgraded monitor may have no insecure clients connected or no clients connected at all, so only the "allowed check" would be raised.  I would expect Rook to upgrade all monitors before taking any action on `auth_allow_insecure_global_id_reclaim` setting.  @travisn is that the case?

**Comment by travisn** [comment]
Rook does have a goroutine that would check every so often if the insecure global IDs can be disabled. When updating the Ceph image, this background routine may be disabling it before all the mons are updated. 

See the check [here](https://github.com/rook/rook/blob/580760c8d38a921620cd06836669724b59fec615/pkg/operator/ceph/cluster/cephstatus.go#L165-L173) where it is disabled. Also see [this comment](https://github.com/rook/rook/issues/7746#issuecomment-859129752) to get it working.

**Comment by agaoglu** [comment]
I'm sorry but, for my case, I just won't race with a goroutine during a live cluster upgrade. I can live with a health warning which I can easily correct after the upgrade.

**Comment by travisn** [comment]
What version of Rook do you have? I'm fuzzy on the details since it was so long ago, but the earlier versions had a softer conversion path. At some point Rook became more strict about enforcing that setting.

That goroutine won't start until after the first cluster reconcile is completed. So if you update the ceph version in the CephCluster CR and immediately restart the Rook operator, Rook will update all the daemons before it will start the goroutine that would possibly disable the insecure clients.

**Comment by agaoglu** [comment]
I'm trying to catch up, starting from rook 1.3 and going on all versions one by one. I thought 1.7 is a good version to upgrade ceph from v14 to v16 since rook 1.8 doesn't support v14. BTW it was backported so it's not really a v16 issue, I just assumed it was.

I might try upgrading ceph in rook 1.5 where it won't know about the globalid reclaim setting. But that will introduce v15 into the mix which I'm trying to avoid if I can.

Following log shows the behaviour:

```
2023-11-29 19:30:48.287511 I | op-mon: deployment for mon rook-ceph-mon-c already exists. updating if needed
2023-11-29 19:30:48.304656 I | op-k8sutil: updating deployment "rook-ceph-mon-c" after verifying it is safe to stop
2023-11-29 19:30:48.304684 I | op-mon: checking if we can stop the deployment rook-ceph-mon-c
2023-11-29 19:31:04.344337 I | op-k8sutil: finished waiting for updated deployment "rook-ceph-mon-c"
2023-11-29 19:31:04.344375 I | op-mon: checking if we can continue the deployment rook-ceph-mon-c
2023-11-29 19:31:04.344395 I | op-mon: waiting for mon quorum with [c a b]
2023-11-29 19:31:04.526318 I | op-mon: mons running: [c a b]
2023-11-29 19:31:09.872195 I | op-mon: Monitors in quorum: [a b]
2023-11-29 19:31:09.885278 I | op-mon: deployment for mon rook-ceph-mon-a already exists. updating if needed
2023-11-29 19:31:09.899138 I | op-k8sutil: updating deployment "rook-ceph-mon-a" after verifying it is safe to stop
2023-11-29 19:31:09.899157 I | op-mon: checking if we can stop the deployment rook-ceph-mon-a
2023-11-29 19:31:10.826736 I | util: retrying after 1m0s, last error: deployment rook-ceph-mon-a cannot be stopped: exit status 16
2023-11-29 19:31:33.571774 I | ceph-cluster-controller: Disabling the insecure global ID as no legacy clients are currently connected. If you still require the insecure connections, see the CVE to suppress the health warning and re-enable the insecure connections. https://docs.ceph.com/en/latest/security/CVE-2021-20288/
```

As you can see, checkStatus jumps while we are waiting to update the second mon.

Anyways, I guess restarting the operator right after patching the CephCluster is doable. Or, maybe update CephCluster CR while operator is offline? Like, stop the operator, patch the CR, then start the operator and let it upgrade stuff without setting anything?

**Comment by travisn** [comment]
Yes, you could update the CR while the operator is offline to ensure it doesn't change the insecure global ID setting before intended. Of course, in case I'm forgetting something, try it in the test cluster first, or maybe you already did by now.

**Comment by agaoglu** [comment]
Yes it's a test cluster and no it didn't work. I guess that's because the goroutine looks like it will start after the very "first cluster reconcile" where rook deploys the cluster, but this is an upgrade so we already have one.

But there's a flag in CR where I can disable that part of monitoring specifically (healthCheck.daemonHealth.status.disabled), which should prevent rook from setting the configuration as it's in the same place. I'll try that on my next attempt.
